src/Account.sol

```diff
@@ -73,70 +66,45 @@ contract Account is IAccount, OpsReady, Auth, Initializable {
-
-    /// @notice helpful modifier to check non-zero values
-    /// @param value: value to check if zero
-    modifier notZero(uint256 value, bytes32 valueName) {
-        if (value == 0) revert ValueCannotBeZero(valueName);
-
-        _;
-    }
+    mapping(uint256 id => ConditionalOrder order) internal conditionalOrders;
 
     /*//////////////////////////////////////////////////////////////
                               CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
     /// @notice disable initializers on initial contract deployment
     /// @dev set owner of implementation to zero address
-    /// @param addressResolver: address of Synthetix ReadProxyAddressResolver
-    /// @param marginAsset: address of Synthetix ProxyERC20sUSD
-    /// @param gelato: address of Gelato
-    /// @param ops: address of Ops
+    /// @param _factory: address of the Smart Margin Account Factory
+    /// @param _events: address of the contract used by all accounts for emitting events
+    /// @param _marginAsset: address of the Synthetix ProxyERC20sUSD contract used as the margin asset
+    /// @param _futuresMarketManager: address of the Synthetix FuturesMarketManager
+    /// @param _gelato: address of Gelato
+    /// @param _ops: address of Ops
     constructor(
-        address addressResolver,
-        address marginAsset,
-        address gelato,
-        address ops
-    ) Auth(address(0)) OpsReady(gelato, ops) {
+        address _factory,
+        address _events,
+        address _marginAsset,
+        address _futuresMarketManager,
+        address _systemStatus,
+        address _gelato,
+        address _ops
+    ) Auth(address(0)) OpsReady(_gelato, _ops) {
         // recommended to use this to lock implementation contracts
         // that are designed to be called through proxies
         _disableInitializers();
 
-        ADDRESS_RESOLVER = IAddressResolver(addressResolver);
-        MARGIN_ASSET = IERC20(marginAsset);
+        FACTORY = IFactory(_factory);
+        EVENTS = IEvents(_events);
+        MARGIN_ASSET = IERC20(_marginAsset);
+        FUTURES_MARKET_MANAGER = IFuturesMarketManager(_futuresMarketManager);
+        SYSTEM_STATUS = ISystemStatus(_systemStatus);
     }
 
-    /// @notice initialize contract (only once) and transfer ownership to specified address
-    /// @dev ensure resolver and sUSD addresses are set to their proxies and not implementations
+    /// @notice initialize contract (only once), transfer ownership to specified address
     /// @param _owner: account owner
-    /// @param _settings: contract address for account settings
-    /// @param _events: address of events contract for accounts
-    /// @param _factory: contract address for account factory
-    function initialize(
-        address _owner,
-        address _settings,
-        address _events,
-        address _factory
-    ) external initializer {
+    function initialize(address _owner) external initializer {
```

Comment:
**Thought:** Just wondering if now that all this does is set the owner if `Initializable` could be removed entirely? And instead perhaps we just used `transferOwnership`. I think you would have to be able to set an `owner` on the `AccountProxy` on construction, so perhaps not worth it. But just a thought.

Reply:
This is an interesting idea. There is a lot of bloat in the `Initializable` contract.
-
You cannot write storage in the constructor of an implementation contract. However, I created a small function that is only callable by the factory, which can set the owner.

lmk what you think

Reply:
I like your solution! Seems good to me. Simplifies and removes a lot of code.